### PR TITLE
feat: lowering replace aten.full_like with aten.full

### PR DESCRIPTION
**Pull request by chohk88**
# Description

1. Lowering: Replaced `aten.full_like` with `aten.full`.
2. Decomposition: For `aten.scatter_add`, replaced `torch.zeros_like` with `torch.zeros`.
3. Removed `expected_ops_param` for the `aten.scatter_add` test case.
4. Linting for `py/torch_tensorrt/dynamo/lowering/_decompositions.py` file.

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Comment by chohk88** [comment]
One concern is that the test case takes a significant amount of runtime, as shown below:

```
DEBUG: [Torch-TensorRT - Debug Build] - Torch-TensorRT TensorRT Engine:
  Name: _run_on_acc_0_engine
  Inputs: [
  ]
  Outputs: [
    id: 0
      name: output0
      shape: [3, 3]
      dtype: Float
  ]
  Device: Device(ID: 0, Name: NVIDIA GeForce RTX 3080 Ti, SM Capability: 8.6, Type: GPU)
  Hardware Compatibility: Disabled

DEBUG: [Torch-TensorRT - Debug Build] - Attempting to run engine (ID: _run_on_acc_0_engine); Hardware Compatible: 0
INFO: [Torch-TensorRT - Debug Build] - Execution profiling is enabled, find results here:
  Device selection profile: /tmp/_run_on_acc_0_engine_device_config_profile.trace
  Input packing profile: /tmp/_run_on_acc_0_engine_input_profile.trace
  Output packing profile: /tmp/_run_on_acc_0_engine_output_profile.trace
  TRT enqueue profile: /tmp/_run_on_acc_0_engine_enqueue_profile.trace
  Engine execution profile: /tmp/_run_on_acc_0_engine_engine_exectuion_profile.trace

DEBUG: [Torch-TensorRT - Debug Build] - Output Name: output0 Shape: [3, 3]
INFO: [Torch-TensorRT - Debug Build] - 
========== _run_on_acc_0_engine profile ==========
                                                   TensorRT layer name    Runtime, %  Invocations  Runtime, ms
                                                                output          0.0%            1         0.00
                   Reformatting CopyNode for Output Tensor 0 to output        100.0%            1         2.28
========== _run_on_acc_0_engine total runtime = 2.27635 ms ==========

INFO: [Torch-TensorRT - Debug Build] - The profiling verbosity was set to ProfilingVerbosity::kLAYER_NAMES_ONLY when the engine was built, so only the layer names will be returned. Rebuild the engine with ProfilingVerbosity::kDETAILED to get more verbose layer information.
..INFO:torch_tensorrt.dynamo.utils:Using Default Torch-TRT Runtime (as requested by user)
INFO:torch_tensorrt.dynamo.utils:Device not specified, using Torch default current device - cuda:0. If this is incorrect, please specify an input device, via the device keyword.
INFO:torch_tensorrt.dynamo.utils:Compilation Settings: CompilationSettings(enabled_precisions={<dtype.f32: 7>}, debug=False, workspace_size=0, min_block_size=1, torch_executed_ops=set(), pass_through_build_failures=True, max_aux_streams=None, version_compatible=False, optimization_level=None, use_python_runtime=False, truncate_double=False, use_fast_partitioner=True, enable_experimental_decompositions=False, device=Device(type=DeviceType.GPU, gpu_id=0), require_full_compilation=False, disable_tf32=False, assume_dynamic_shape_support=False, sparse_weights=False, make_refitable=False, engine_capability=<EngineCapability.STANDARD: 1>, num_avg_timing_iters=1, dla_sram_size=1048576, dla_local_dram_size=1073741824, dla_global_dram_size=536870912, dryrun=False, hardware_compatible=False, timing_cache_path='/tmp/timing_cache.bin', lazy_engine_init=False)
```

I think we could optimize here  https://github.com/pytorch/TensorRT/blob/6321710e645bfed15f8bcfe3558406dbbe707ce3/py/torch_tensorrt/dynamo/conversion/impl/full.py#L28

**Comment by peri044** [comment]
> One concern is that the test case takes a significant amount of runtime, as shown below:
> 
> ```
> DEBUG: [Torch-TensorRT - Debug Build] - Torch-TensorRT TensorRT Engine:
>   Name: _run_on_acc_0_engine
>   Inputs: [
>   ]
>   Outputs: [
>     id: 0
>       name: output0
>       shape: [3, 3]
>       dtype: Float
>   ]
>   Device: Device(ID: 0, Name: NVIDIA GeForce RTX 3080 Ti, SM Capability: 8.6, Type: GPU)
>   Hardware Compatibility: Disabled
> 
> DEBUG: [Torch-TensorRT - Debug Build] - Attempting to run engine (ID: _run_on_acc_0_engine); Hardware Compatible: 0
> INFO: [Torch-TensorRT - Debug Build] - Execution profiling is enabled, find results here:
>   Device selection profile: /tmp/_run_on_acc_0_engine_device_config_profile.trace
>   Input packing profile: /tmp/_run_on_acc_0_engine_input_profile.trace
>   Output packing profile: /tmp/_run_on_acc_0_engine_output_profile.trace
>   TRT enqueue profile: /tmp/_run_on_acc_0_engine_enqueue_profile.trace
>   Engine execution profile: /tmp/_run_on_acc_0_engine_engine_exectuion_profile.trace
> 
> DEBUG: [Torch-TensorRT - Debug Build] - Output Name: output0 Shape: [3, 3]
> INFO: [Torch-TensorRT - Debug Build] - 
> ========== _run_on_acc_0_engine profile ==========
>                                                    TensorRT layer name    Runtime, %  Invocations  Runtime, ms
>                                                                 output          0.0%            1         0.00
>                    Reformatting CopyNode for Output Tensor 0 to output        100.0%            1         2.28
> ========== _run_on_acc_0_engine total runtime = 2.27635 ms ==========
> 
> INFO: [Torch-TensorRT - Debug Build] - The profiling verbosity was set to ProfilingVerbosity::kLAYER_NAMES_ONLY when the engine was built, so only the layer names will be returned. Rebuild the engine with ProfilingVerbosity::kDETAILED to get more verbose layer information.
> ..INFO:torch_tensorrt.dynamo.utils:Using Default Torch-TRT Runtime (as requested by user)
> INFO:torch_tensorrt.dynamo.utils:Device not specified, using Torch default current device - cuda:0. If this is incorrect, please specify an input device, via the device keyword.
> INFO:torch_tensorrt.dynamo.utils:Compilation Settings: CompilationSettings(enabled_precisions={<dtype.f32: 7>}, debug=False, workspace_size=0, min_block_size=1, torch_executed_ops=set(), pass_through_build_failures=True, max_aux_streams=None, version_compatible=False, optimization_level=None, use_python_runtime=False, truncate_double=False, use_fast_partitioner=True, enable_experimental_decompositions=False, device=Device(type=DeviceType.GPU, gpu_id=0), require_full_compilation=False, disable_tf32=False, assume_dynamic_shape_support=False, sparse_weights=False, make_refitable=False, engine_capability=<EngineCapability.STANDARD: 1>, num_avg_timing_iters=1, dla_sram_size=1048576, dla_local_dram_size=1073741824, dla_global_dram_size=536870912, dryrun=False, hardware_compatible=False, timing_cache_path='/tmp/timing_cache.bin', lazy_engine_init=False)
> ```
> 
> I think we could optimize here
> 
> https://github.com/pytorch/TensorRT/blob/6321710e645bfed15f8bcfe3558406dbbe707ce3/py/torch_tensorrt/dynamo/conversion/impl/full.py#L28

Do you mean the static case which uses np.full takes more time or the dynamic one ? For the static, I'm not sure what control we have for optimization. Also, there would be variation b/w debug and release build. So, you might want to try the release build for perf measurement.

**Comment by peri044** [comment]
@chohk88 Opened a new issue for the perf improvement you were mentioning https://github.com/pytorch/TensorRT/issues/3107
